src/v2/cookbook/unit-testing-vue-components.md

```diff
@@ -0,0 +1,234 @@
+---
+title: Unit Testing Vue Components
+type: cookbook
+order: 1.2
+---
+
+## Simple Example
+
+Unit testing is a fundamental part of software developmenet. The underlying idea is tests the smallest units of code, in isolation from others. This makes it easy to refactor, add new features, or track down bugs. Vue's [single-file components](./single-file-components.html), it is straight forward to write unit tests for components in isolation. This lets you develop new features with confidence you are not breaking existing ones, and helps other people to understand what your component does.
+
+This simple example tests whether some text is rendered:
+
+```html
+<template>
+  <div>
+    <input v-model="username">
+    <div 
+      v-if="error"
+      class="error"
+    >
+      {{ error }}
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'Hello',
+  data () {
+    return {
+      username: ''
+    }
+  },
+
+  computed: {
+    error () {
+      return this.username.length < 7
+        ? 'Please enter a longer username'
+        : ''
+    }
+  }
+}
+</script>
+```
+
+```js
+import { shallow } from 'vue-test-utils'
+
+test('Foo', () => {
+  // render the component
+  const wrapper = shallow(Hello)
+
+  // assert the error is rendered
+  expect(wrapper.find('error').exists()).toBe(true)
+
+  // update the name to be long enough
+  wrapper.setData({
+    username: {
+      'Lachlan'
+    }
+  })
+
+  // assert the error has gone away
+  expect(wrapper.find('error').exists()).toBe(false)
+})
+```
+
+This simple example shown how to test whether an error message is rendered based on the length of the username. It demonstrates the general idea of unit testing Vue components: render the component, and make assertions to check if the markup matches the state of the component.
+
+## Details about the Value
```

Comment:
I don't think you need to literally say "details about the value, I think you could change the h2 directly to "Why Test"